internal/store/store.go

```diff
@@ -10,19 +14,68 @@ import (
 
 // Store provides access to a store backed by the given reader.
 type Store struct {
-	plugins []*model.Plugin
+	plugins []pluginVersions
 	logger  logrus.FieldLogger
 }
 
+// pluginVersions is a sorted list of all versions of a given plugin.
```

Comment:
nit: worth defining this before `Store` to read top-to-bottom? But actually, is this type adding a lot of value? I'm stumbling over `[]pluginVersions`, trying to remember what it means, when perhaps `[][]*model.Plugin` would be clearer?